base/tuple.jl

```diff
@@ -160,7 +160,8 @@ heads(t::Tuple, ts::Tuple...) = (t[1], heads(ts...)...)
 tails() = ()
 tails(t::Tuple, ts::Tuple...) = (tail(t), tails(ts...)...)
 map(f, ::Tuple{}, ts::Tuple...) = ()
-map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (f(heads(t1, t2, ts...)...), map(f, tails(t1, t2, ts...)...)...)
+map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (@_inline_meta;
```

Comment:
Does this definition not result in arbitrarily deep inlining in the same manner that [the two argument form](https://github.com/JuliaLang/julia/blob/474d05d0bef8bd2fad984db0223162de922f0315/base/tuple.jl#L145) does without [this additional definition]( https://github.com/JuliaLang/julia/blob/474d05d0bef8bd2fad984db0223162de922f0315/base/tuple.jl#L149) that limits inlining?